allennlp/models/encoder_decoders/wikitables_semantic_parser.py

```diff
@@ -271,6 +310,166 @@ def forward(self,  # type: ignore
+        is accepted as a parameter for copying purposes
+
+        Parameters
+        ----------
+        worlds : ``List[WikiTablesWorld]``
+        num_entities : ``int``
+        tensor : ``torch.Tensor``
+            Used for copying the constructed list onto the right device.
+
+        Returns
+        -------
+        A ``torch.autograd.Variable`` with shape ``(batch_size, num_entities, num_entities)``.
+        Dimension 2 contains the indices for the neighbors of the entity represented by the
+        index of dimension 1. Dimension 2 is limited to num_entities as it is assumed that
+        each entity can have a number of neighbors at most equal to the total number of entities.
+        """
+        batches_neighbors = []
+        for world in worlds:
+            # Each batch instance has its own world, which has a corresponding table.
+            entities = world.table_graph.entities
+            entity2indices = {entity: i for i, entity in enumerate(entities)}
+            entity2neighbors = world.table_graph.neighbors
+            neighbor_indexes = []
+            for entity in entities:
+                entity_neighbors = [entity2indices[cn] for cn in entity2neighbors[entity]]
+                padded = pad_sequence_to_length(entity_neighbors, num_entities)
+                neighbor_indexes.append(padded)
+            neighbor_indexes = pad_sequence_to_length(neighbor_indexes,
+                                                      num_entities,
+                                                      lambda: [0] * num_entities)
+            batches_neighbors.append(neighbor_indexes)
+        return Variable(tensor.data.new(batches_neighbors)).long()
+
+    @staticmethod
+    def _get_type_vector(worlds: List[WikiTablesWorld],
+                         num_entities: int,
+                         tensor: torch.Tensor) -> Tuple[Variable, Dict[int, int]]:
+        """
+        Produces the one hot encoding for each entity's type. In addition,
+        a map from index to type is returned to combine entity type operations
+        into one method.
+
+        Parameters
+        ----------
+        worlds : ``List[WikiTablesWorld]``
+        num_entities : ``int``
+        tensor : ``torch.Tensor``
+            Used for copying the constructed list onto the right device.
+
+        Returns
+        -------
+        A ``torch.autograd.Variable`` with shape ``(batch_size, num_entities, num_types)``.
+        entity_types : ``Dict[int,str]``
+            This is a mapping from (batch_index, entity_index) to entity type id.
+        """
+        entity_types = {}
+        batch_types = []
+        for batch_index, world in enumerate(worlds):
+            types = []
+            for entity_index, entity in enumerate(world.table_graph.entities):
+                cell_type_one_hot = [1, 0]
+                row_type_one_hot = [0, 1]
+                types.append((cell_type_one_hot if entity.startswith('fb:cell') else row_type_one_hot))
```

Comment:
You should only have the type check once.  If you move the `entity_type = 0 ...` line up here, you can just do `types.append([1, 0] if entity_type == 0 else [0, 1])`.